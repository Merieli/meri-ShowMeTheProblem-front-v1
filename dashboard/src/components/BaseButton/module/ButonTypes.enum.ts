export enum ButtonTypes {
    BUTTON = 'button',
    SUBMIT = 'submit',
}

export type TButton = ButtonTypes.BUTTON | ButtonTypes.SUBMIT
