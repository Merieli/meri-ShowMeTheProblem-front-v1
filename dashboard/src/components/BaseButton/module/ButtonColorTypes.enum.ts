export type TColorsButton = 'light' | 'dark';
