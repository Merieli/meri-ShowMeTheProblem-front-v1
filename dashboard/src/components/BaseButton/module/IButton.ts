import { TButton } from './ButonTypes.enum';
import { TColorsButton } from './ButtonColorTypes.enum';

export interface IBaseButton {
    color: TColorsButton;
    type: TButton;
}
