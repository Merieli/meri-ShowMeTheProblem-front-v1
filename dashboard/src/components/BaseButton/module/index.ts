export * from './ButonTypes.enum';
export * from './ButtonColorTypes.enum';
export * from './IButton';
