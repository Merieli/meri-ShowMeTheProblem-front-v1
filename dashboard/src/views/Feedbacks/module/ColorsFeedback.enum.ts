export enum ColorsFeedback {
    ALL = 'brand-info',
    ISSUE = 'brand-danger',
    IDEA = 'brand-warning',
    OTHER = 'brand-graydark',
}
