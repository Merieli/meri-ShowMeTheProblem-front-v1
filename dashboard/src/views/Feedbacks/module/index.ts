export * from './Labels.enum'
export * from './ColorsFeedback.enum'
