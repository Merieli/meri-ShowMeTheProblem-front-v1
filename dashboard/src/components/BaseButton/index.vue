<template>
    <button
        class="base-button rounded-full font-bold text-xl"
        :class="classByColor"
        :type="typeButton"
        data-button="base"
    >
        <!-- @slot texto para o botão -->
        <slot />
    </button>
</template>

<script lang="ts">
/**
 * Botão base que pode ser utilizado em qualquer parte do projeto.
 * @displayName Base Button
 * @author Merieli Manzano
 * @version 1.0.0
 */
import { ButtonColors, ButtonTypes, TColorsButton, TButton } from './module'
import { defineComponent, computed, ComputedRef, PropType } from 'vue'

export default defineComponent({
    name: 'BaseButton',
    props: {
        /**
         * Define a cor de **fundo** e de **fonte** do botão.
         * @values light, dark
         */
        color: {
            type: String as PropType<TColorsButton>,
            default: ButtonColors.LIGHT,
        },
        /**
         * Define se o botão é normal ou é um botão de formulário.
         * @values button, submit
         */
        typeButton: {
            type: String as PropType<TButton>,
            default: ButtonTypes.BUTTON,
        },
    },
    setup(props) {
        /**
         * Define a classe que será utilizada no botão de acordo com sua **color** definida.
         * @method classByColor
         * @type {computed}
         */
        const classByColor: ComputedRef<string> = computed(() => {
            return props.color === ButtonColors.DARK
                ? 'base-button_dark'
                : ButtonColors.LIGHT
                ? 'base-button_light'
                : 'base-button_dark'
        })

        return {
            classByColor,
        }
    },
})
</script>

<style lang="scss" scoped>
.base-button {
    padding: 7px 22px;
    max-width: 250px;
    border: none;
    border-radius: 5rem;

    &_dark {
        background-color: theme('colors.brand.main');
        color: theme('colors.brand.white');
    }

    &_light {
        background-color: theme('colors.brand.white');
        color: theme('colors.brand.main');
    }
}
</style>

<docs lang="md">
Botão base que pode ser utilizado em qualquer parte do projeto.

## Exemplos

### Botão Dark:

```jsx
<BaseButton color="dark" type-button="submit">
    Efetuar login
</BaseButton>
```

Botão Light:

```jsx
<BaseButton color="light"> Teste </BaseButton>
```
</docs>
