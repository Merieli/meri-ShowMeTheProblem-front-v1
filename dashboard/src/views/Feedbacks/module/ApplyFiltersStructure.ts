import { IFeedbackFilters, IConfiguredFilters } from '@/interfaces'
import { feedbackFilterColors, feedbackFilterLabels } from '@/views/Feedbacks/module'

export const applyFiltersStructure = (summary: IFeedbackFilters, filterActive?: string) => {
    return Object.keys(summary).map((typeOfFeedback) => {
        const checkActive = (): boolean => {
            if (typeOfFeedback === 'all' && filterActive == undefined) return true

            if (filterActive != undefined && typeOfFeedback === filterActive) return true

            return false
        }

        const currentFilter: IConfiguredFilters = {
            label: feedbackFilterLabels[typeOfFeedback as keyof IFeedbackFilters],
            color: feedbackFilterColors[typeOfFeedback as keyof IFeedbackFilters],
            amount: summary[typeOfFeedback as keyof IFeedbackFilters],
            active: checkActive(),
            type: typeOfFeedback,
        }

        return currentFilter
    })
}
