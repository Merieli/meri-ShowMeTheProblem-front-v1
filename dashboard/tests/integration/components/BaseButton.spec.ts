import BaseButton from '@/components/BaseButton/index.vue'
import { ButtonColors, ButtonTypes } from '@/components/BaseButton/module'
import { mount, VueWrapper } from '@vue/test-utils'

describe('BaseButton', () => {
    let wrapper: VueWrapper

    beforeEach(() => {
        wrapper = mount(BaseButton, {
            slots: {
                default: 'Ação X',
            },
            props: {},
        })
    })

    afterEach(() => {
        wrapper.unmount()
    })

    describe('Integração/Componente', () => {
        describe('Renderização:', () => {
            test('Dado o componente quando for montado então deve ser uma instancia vue', async () => {
                expect(wrapper.vm).toBeDefined()
            })

            test('Dado o preenchimento do slot quando definido um texto "Ação X" então exibe esse texto no componente', async () => {
                const button = wrapper.find('button.base-button')

                expect(button.html()).toContain('Ação X')
            })

            test('Dado a prop color quando definida como "dark" então possui a classe "base-button_dark"', async () => {
                await wrapper.setProps({ color: ButtonColors.DARK })

                expect(wrapper.classes()).toContain('base-button_dark')
            })

            test('Dado a prop color quando definida como "light" então possui a classe "base-button_light"', async () => {
                await wrapper.setProps({ color: ButtonColors.LIGHT })

                expect(wrapper.classes()).toContain('base-button_light')
            })

            test('Dado a prop typeButton quando definida como "submit" então o botão deve ter o type "submit"', async () => {
                await wrapper.setProps({ typeButton: ButtonTypes.SUBMIT })

                expect(wrapper.attributes('type')).toContain(ButtonTypes.SUBMIT)
            })
        })
    })
})
