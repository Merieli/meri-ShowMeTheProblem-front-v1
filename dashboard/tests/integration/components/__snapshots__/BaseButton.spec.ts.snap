// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton Integração/Componente Renderização: Dado o html quando renderizado então deve ter os mesmos dados do snapshot gravado 1`] = `
"<button class=\\"base-button rounded-full font-bold text-xl transition-all duration-150 base-button_light\\" type=\\"button\\" data-button=\\"base\\">
  <!-- @slot Texto de conteúdo do botão -->Ação X
</button>"
`;
